app/models/hmpps_api/prison_timeline.rb

```diff
@@ -0,0 +1,24 @@
+# frozen_string_literal: true
+
+# This model represents a persons journey through their various prison visits - its goal is to tell
+# us which prisons they were in and when
+module HmppsApi
+  class PrisonTimeline
+    def initialize(movements)
+      @movements = movements.reverse
+    end
+
+    def last_movement
+      @movements.first
+    end
+
+    # obtain a unique visit number for the date specified
+    def prison_visit date
+      @movements.each_with_index.map { |m, index| [m, index] }.detect { |m, _index| m.movement_date <= date }.second
+    end
+
```

Comment:
My only comment @starswan: I think I might have felt a bit muddled with this name(prison_visit) as it feels very fleeting and more likely feels like someone visiting a prisoner. Wouldn't it be something more substantial than a visit like a prison_stay or prison_resides